base/src/com/google/idea/blaze/base/sync/sharding/BlazeBuildTargetSharder.java

```diff
@@ -111,6 +109,9 @@ private enum ShardingApproach {
 
   private static ShardingApproach getShardingApproach(
       SyncStrategy parallelStrategy, ProjectViewSet viewSet) {
+    if (parallelStrategy == SyncStrategy.SERIAL_NOT_EXPAND) {
```

Comment:
So this overrides the `shard_sync` setting in the PV, how about just keeping the old behavior here?

Reply:
The issue here is that compile folder with shard_sync=true takes sooo much more time. For example, one of our project's folder compiles in 17s with shard_sync=true and in 1s with shard_sync=false. The main reason why we added this button was that sync time was very slow, and we wanted to speed up compile errors check. Ordinary user won't set shard_sync to false. We can add another setting to set shard_sync for build folder action, but I think in cases when OOM happens, you can just use Partial Sync instead

Reply:
Oh, that's a good point. Btw what do you think about setting the default PV in `tools/intellij/.managed.bazelproject` with `shard_sync: true`?.

Example here https://github.com/bazelbuild/bazel/blob/master/tools/intellij/.managed.bazelproject

Reply:
Personally, I think It's a good idea, but I don't have a lot of knowledge about how often people want to use `shard_sync=true`. In our project we use it quite rarely, but I can't expand my PV on all users of the plugin. 
Should I create another PR [here](https://github.com/bazelbuild/bazel/blob/master/tools/intellij/.managed.bazelproject) or we can stay on current realization or mb another ideas?

Reply:
@LeFrosch what do you think about it?

Reply:
Oh, actually if I am not mistaken, `shard-sync=true` by default. We can delete this property in enum

Reply:
Deleted SERIAL_NOT_EXPAND. Sorry for disappointing, that's my bad

Reply:
But anyway logically it would be better if `shard-sync` won't affect build. Often`shard-sync=true` is used for full sync of project if OOM happens. `shard-sync=true` will affect the build folder speed. (17s and 1s is big diff) I don't think that it will be used for the whole project often. With `shard-sync=true` build folder action is useless, because the diff between build and sync time isn't big. I think SERIAL_NOT_EXPAND anyway will be better

Reply:
Okay, but I can't guarantee it will be permanent. If it turns out to be misleading, we’ll adjust it to align with the shard_sync setting.